### PR TITLE
load fewer messages from disk when serving blocksync requests

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Review by schomatis — APPROVED** [review]

**Comment by schomatis** [comment]
Not sure what's the test error.

**Review by magik6k — APPROVED** [review]

**Review by vyzo — APPROVED** [review]

**Review by schomatis — CHANGES_REQUESTED** [review]
The `TestSyncSimple` is stalled (tested locally). I can review this in detail later today if you can't fix.

File: chain/blocksync/server.go [hunk 1]
```diff
@@ -221,43 +221,52 @@ func collectChainSegment(
...
 
 			bmi = append(bmi, i)
 		}
 		blsincl = append(blsincl, bmi)
 
-		smi := make([]uint64, 0, len(smsgs))
-		for _, m := range smsgs {
-			i, ok := secpkmsgmap[m.Cid()]
+		smi := make([]uint64, 0, len(sc))
+		for _, m := range secpkcids {
```
Comment (magik6k) [inline chain/blocksync/server.go:248]:
```suggestion
		for _, m := range sc {
```

**Review by schomatis — APPROVED** [review]
Now we're talking, thanks @magik6k.

**Pull request merged**